docs/detections/rules-ui-create.asciidoc

```diff
@@ -358,6 +360,30 @@ FROM logs-* [metadata _id, _index, _version]
 | LIMIT 10
 -----
 
+[float]
+[[esql-query-design]]
+===== Query design considerations 
+
+When writing your query, consider the following:
+
+- The {ref}/esql-commands.html#esql-limit[`LIMIT`] command specifies the number of rows an {esql} query returns and the number of alerts created per rule execution. Similarly, the <<opt-fields-all,`max_signals`>> setting specifies how many alerts a rule can create every time it runs.
++ 
+If the `LIMIT` value is lower than the `max_signals` value, the rule uses the `LIMIT` value. If the `LIMIT` is higher than `max_signals`, the rule uses `max_signals`. 
++
+NOTE: The `max_signals` default value is 100. You can modify it using the <<rules-api-create,Create rule API>>. 
++
+
+- When writing an aggregating query, use the {ref}/esql-commands.html#esql-keep[`KEEP`] command to keep fields that you want to examine. For example, if you want to search and filter for the `host.name`, `user.name`, and `process.name` fields, keep them in your query. 
```

Comment:
This won't work. Aggregating query returns only fields defined after ~BY~ keyword, apart from field that has value of aggregation

https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-commands.html#esql-stats-by

> The STATS ... BY processing command groups rows according to a common value and calculate one or more aggregated values over the grouped rows. If BY is omitted, the output table contains exactly one row with the aggregations applied over the entire dataset.

So users will need to use these fields in `BY` clause of the command

Reply:
@vitaliidm thanks for the check here - how should this be re-written?

Reply:
I would say 
```suggestion
- When writing an aggregating query, if you want to search and filter for the `host.name`, `user.name`, `process.name` fields or any other field, place them after {ref}/esql-commands.html#esql-stats-by[`BY operator`]  command.
```

Reply:
>  For example, placing the `host.name`, `user.name`, `process.name` fields after the BY operator of `STATS...BY` command returns these fields in alert documents, and allows you to search and filter for them from the Alerts table.

Reply:
```suggestion
- When writing an aggregating query, follow the {ref}/esql-commands.html#esql-stats-by[`STATS...BY`] command with fields that you want to search and filter for when alerts are created. For example, placing the `host.name`, `user.name`, `process.name` fields after the `BY` operator of the `STATS...BY` command returns these fields in alert documents, and allows you to search and filter for them from the Alerts table. 
```